lib/src/domain/usecase/implementation/search_by_title_usecase.dart

```diff
@@ -0,0 +1,27 @@
+import '../../../core/parameter/search_parameters.dart';
+import '../../../core/util/data_state.dart';
+import '../../../core/util/enums.dart';
+import '../../../data/repository/movie_repository.dart';
+import '../../entity/movie.dart';
+import '../usecase_interface.dart';
+
+class SearchByTitleUseCase
```

Comment:
Review the UseCases, all are different in terms of the endpoint that you use, but you always are returning the same data, so maybe you can create only one UseCase